pkg/scheduler/factory.go

```diff
@@ -182,22 +182,50 @@ func (c *Configurator) create() (*Scheduler, error) {
 	)
 	debugger.ListenForSignal(c.StopEverything)
 
-	algo := NewGenericScheduler(
+	sched := newScheduler(
 		c.schedulerCache,
+		extenders,
+		profiles,
+		internalqueue.MakeNextPodFunc(podQueue),
+		MakeDefaultErrorFunc(c.client, c.informerFactory.Core().V1().Pods().Lister(), podQueue, c.schedulerCache),
+		c.StopEverything,
+		podQueue,
 		c.nodeInfoSnapshot,
 		c.percentageOfNodesToScore,
-	)
+		nil)
```

Comment:
All Changes done in commit-2. 😀

Reply:
Thanks, my suggestion was to remove the assignment at lines https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L283-L284 and pass in the client and stopEverything to newScheduler so that setting up the sched struct is done in one place.

Reply:
Hi, @ahg-g 
In commit-3, I moved initialization of `client` and `stopEverything` into `newScheduler,` 
and adjusted the parameter list of `newScheduler()` in the same order to struct `Scheduler`.